frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -628,10 +606,11 @@ void Visitor::checkNoReceiverClauseOnPrimaryMethod(const Function* node) {
           receiverTypeStr = ss.str();
         }
 
-        error(node, "type binding clauses ('%s.' in this case) are not "
-                    "supported in declarations within a class, record "
-                    "or union.",
-                    receiverTypeStr.c_str());
+        CHPL_POSTPARSE_ERR_SIMPLE(
+            builder_, node,
+            "type binding clauses ('" + receiverTypeStr +
```

Comment:
All of these string operations make me think that we want to support varargs / printf style for general postparse errors etc. Less string copying, cleaner code.

Reply:
I agree. The previous error reporting mechanism here (which was specific to this file) was a function with `printf` style string interpolation. I'll bring something like that back.